src/modules/universal-discovery/components/content-table/content.table.item.component.js

```diff
@@ -1,33 +1,48 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
+import SelectContentButtonComponent from '../select-content-button/select.content.button.component';
+
 import './css/content.table.item.component.css';
 
-const ContentTableItemComponent = (props) => {
-    const { onItemClick, onPreview, data, contentTypesMap, labels } = props;
-    const item = data.ContentInfo.Content;
-    const contentType = contentTypesMap ? contentTypesMap[item.ContentType._href] : false;
-    const contentTypeName = contentType ? contentType.names.value[0]['#text'] : labels.contentTableItem.notAvailable;
-    const onClick = !!onItemClick ? onItemClick.bind(null, data) : null;
+export default class ContentTableItemComponent extends Component {
```

Comment:
You should go back to stateless component. The `renderSelectContentBtn` can be defined outside of `const ContentTableItemComponent` definition, just pass the props to it.

Reply:
Good catch. During development I changed this to component with state and then removed the state :)